platform/osgi/platform-osgi-configadmin/src/main/java/org/codice/felix/cm/file/DelegatingPersistenceManager.java

```diff
@@ -108,28 +194,17 @@ private static BundleContext getBundleContext() {
     return FrameworkUtil.getBundle(DelegatingPersistenceManager.class).getBundleContext();
   }
 
-  private static class ConfigInterceptingCustomizer
+  /**
+   * Service callbacks.
+   */
+  class PluginTrackerCustomizer
       implements ServiceTrackerCustomizer<
           ConfigurationPersistencePlugin, ConfigurationPersistencePlugin> {
-    private final ConfigurationContextFactory factory = new ConfigurationContextFactoryImpl();
-    private final StampedLock configDeadZoneLock;
-
-    ConfigInterceptingCustomizer(StampedLock configDeadZoneLock) {
-      this.configDeadZoneLock = configDeadZoneLock;
-    }
 
     @Override
     public ConfigurationPersistencePlugin addingService(
         ServiceReference<ConfigurationPersistencePlugin> serviceReference) {
-      ConfigurationPersistencePlugin plugin = getBundleContext().getService(serviceReference);
-      // Write lock because no configs should be getting saved or deleted while we initialize
-      long stamp = configDeadZoneLock.writeLock();
-      try {
-        plugin.initialize(factory);
-      } finally {
-        configDeadZoneLock.unlockWrite(stamp);
-      }
-      return plugin;
+      return handleServiceAdded(serviceReference);
```

Comment:
✏️  At this point, does it not make sense to just have `DelegatingPersistenceManager` implement the `ServiceTrackerCustomizer` interface? I'm saying that because this inner class literally just delegates to a method on the outer class.

Reply:
Originally that's how I had it, but it requires the pass of the `this` pointer in the constructor of delegating persistence manager. If an exception gets thrown we could have a floating reference to a partially constructed object.